generate-types/index.js

```diff
@@ -1847,9 +1847,10 @@ const printError = (diagnostic) => {
 							code: getCode(arg.type, innerTypeArgs),
 							optional: false,
 					  };
-				canBeOptional = canBeOptional && optional;
+				const isCurrentOptional = optional && !arg.spread;
```

Comment:
there is a case in types where spread argument is optional, not sure why right now all works